tests/test_bases.py

```diff
@@ -42,3 +99,24 @@ def test_get_root_lengths(canola_h5):
     np.testing.assert_array_almost_equal(
         root_lengths, [20.129579, 62.782368, 80.268003, 34.925591, 3.89724]
     )
+
+
+def test_get_root_lengths_no_roots():
+    pts = np.array(
+        [
+            [
+                [np.nan, np.nan],
+                [np.nan, np.nan],
+            ],
+            [
+                [np.nan, np.nan],
+                [np.nan, np.nan],
+            ],
+        ]
+    )
+
+    assert pts.shape == (2, 2, 2)
```

Comment:
This assertion is a bit unnecessary since the shape is clear from the array definition above. You mainly want to use it when doing other intermediate operations on the test data

Reply:
I removed the assertion.